server/src/main/java/org/elasticsearch/index/mapper/ObjectRuntimeField.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.script.ObjectFieldScript;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A runtime field of type object. Defines a script at the top level, which emits multiple sub-fields.
+ * The sub-fields are declared within the object in order to be made available to the field_caps and search API.
+ */
+public class ObjectRuntimeField implements RuntimeField {
+
+    public static final Parser PARSER = new Parser(name ->
+        new RuntimeField.Builder(name) {
+            private final FieldMapper.Parameter<Script> script = new FieldMapper.Parameter<>(
+                "script",
+                false,
+                () -> null,
+                AbstractScriptFieldType.Builder::parseScript,
+                RuntimeField.initializerNotSupported()
+            ).setValidator(s -> {
```

Comment:
it seems like this validator does nothing, I need to look into it. @romseygeek any idea?

Reply:
It will throw an error if there's no script defined on the object field, no?

Reply:
that's what I'd expect it to do, but I meant to bring up that it has no effect.

Reply:
I tracked it. we no longer call validate on the parameters. It is no problem as we never use a validator in the existing parameters. I will fix this